torch/_meta_registrations.py

```diff
@@ -169,7 +169,10 @@ def meta_linalg_qr_helper(input, mode):
     R.transpose_(-2, -1)
     return (Q, R)
 
-@torch.library.impl(meta_lib, "index.Tensor")
+# Leaving this function around because a python implementation
+# of indexing shape inference is useful,
+# but not registering it to the dispatcher because we already
+# get shape inference through structured kernels
```

Comment:
@ezyang let me know if you want to keep this around. It seems useful to have a python impl, but we can't register it now that `index.Tensor` is structured without some other hacks (we'll end up with a duplicate registration)

Reply:
Yeah just keep it without registering